### PR TITLE
[IRRC-75 (part 2)] refactoring new teacher for vuex

### PR DESCRIPTION
**Pull request by lauratsui**
made changes in NewTeacher.vue for vuex

### CODE REVIEW

File: frontend/src/pages/NewTeacher.vue [hunk 1]
```diff
@@ -1,352 +1,350 @@
...
             </section>
-            <b-button class="dark-blue" input type="submit" expanded @click="createTeacher" :disabled="formIsInvalid">Create Student</b-button>
+            <b-button
+              class="dark-blue"
+              input
+              type="submit"
+              expanded
+              @click="createNewTeacher"
+              :disabled="formIsInvalid"
+              >Create Student</b-button
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue]:
```suggestion
              >Create Teacher</b-button
```

File: frontend/src/pages/NewTeacher.vue [hunk 2]
```diff
@@ -1,352 +1,350 @@
...
           <img src="../assets/teacher.png" />
         </div>
-        <!-- Start of 2nd column (all input fields) --> 
+        <!-- Start of 2nd column (all input fields) -->
         <div class="column is-two-thirds">
-          <form method="POST" action ="/api/students" @submit.prevent="createTeacher">
-          <!--.prevent prevents the default submit behaviour and executes createTeacher instead -->
+          <form
+            method="POST"
+            action="/api/students"
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue]:
```suggestion
            action="/api/teachers"
```

File: frontend/src/pages/NewTeacher.vue [hunk 3]
```diff
@@ -1,352 +1,350 @@
...
+                    <option value="Little">
+                      Little (Able to understand simple words)
+                    </option>
+                    <option value="Simple">
+                      Simple (Able to speak full sentences)
+                    </option>
+                    <option value="Intermediate">
+                      Intermediate (Able to understand simple words)
+                    </option>
+                  </b-select>
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue:132]: TODO: update these options as specified in https://github.com/TechLadies/irrc-madwish/issues/85

File: frontend/src/pages/NewTeacher.vue [hunk 4]
```diff
@@ -1,352 +1,350 @@
...
+                  >
+                    <option value="No">No (Unable to understand at all)</option>
+                    <option value="Little">
+                      Little (Able to understand simple words)
+                    </option>
+                    <option value="Simple">
+                      Simple (Able to speak full sentences)
+                    </option>
+                    <option value="Intermediate">
+                      Intermediate (Able to understand simple words)
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue:88]: TODO: update these options as specified in https://github.com/TechLadies/irrc-madwish/issues/85

**Review by therealyingtong — CHANGES_REQUESTED** [review]

File: frontend/src/pages/NewTeacher.vue [hunk 5]
```diff
@@ -1,352 +1,350 @@
...
-                  <span class="file-cta">
-                      <b-icon class="file-icon" icon="upload"></b-icon>
-                      <span class="file-label">  Bulk upload CSV</span>
-                  </span>
-                  <span class="file-name" v-if="file">
-                      {{ file.name }}
-                  </span>
-              </b-upload>
-          </b-field>   
+        <b class="newstudent">New Teacher</b>
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue]:
```suggestion
        <b class="newteacher">New Teacher</b>
```

File: frontend/src/pages/NewTeacher.vue [hunk 6]
```diff
@@ -1,352 +1,350 @@
...
+      languages: [],
+      nativeLanguage: "",
+    };
   },
   computed: {
-    ...mapGetters(['API_nativeLanguage']),
-
+    ...mapGetters(["API_nativeLanguage"]),
 
     // Checks if required fields are empty. If required fields are empty, the Create Student Button is disabled.
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue]:
```suggestion
    // Checks if required fields are empty. If required fields are empty, the Create Teacher Button is disabled.
```

File: frontend/src/pages/NewTeacher.vue [hunk 7]
```diff
@@ -1,352 +1,350 @@
...
   border-radius: 4px;
 }
 
-
 .label {
-  color: #59666E !important;
+  color: #59666e !important;
 }
 
 b.newstudent {
```
Comment (therealyingtong) [inline frontend/src/pages/NewTeacher.vue]:
```suggestion
b.newteacher {
```

**Review by therealyingtong — CHANGES_REQUESTED** [review]

File: frontend/src/pages/NewTeacher.vue (hunk 4 above)
Comment (lauratsui) [reply frontend/src/pages/NewTeacher.vue:88]: Noted, will update as part of a separate PR as discussed

File: frontend/src/pages/NewTeacher.vue (hunk 3 above)
Comment (lauratsui) [reply frontend/src/pages/NewTeacher.vue:132]: Noted, will update as part of a separate PR as discussed

**Review by therealyingtong — APPROVED** [review]

**Review by xuanjane — APPROVED** [review]
looks good :)

**Pull request merged**